codegen-sbt/src/main/scala/caliban/codegen/CodegenPlugin.scala

```diff
@@ -12,7 +12,11 @@ object CodegenPlugin extends AutoPlugin {
   lazy val genSchemaCommand         = genCommand("calibanGenSchema", genSchemaHelpMsg, SchemaWriter.write)
   lazy val genClientCommand         = genCommand("calibanGenClient", genClientHelpMsg, ClientWriter.write)
 
-  def genCommand(name: String, helpMsg: String, writer: (Document, String, Option[String], String) => String): Command =
+  def genCommand(
+    name: String,
+    helpMsg: String,
+    writer: (Document, String, Option[String], String, Map[String, String]) => String
```

Comment:
Hmm, this tuple is becoming more and more cryptic, how about using a case class?